base/libgit2/status.jl

```diff
@@ -14,12 +14,11 @@ function Base.length(status::GitStatus)
 end
 
 function Base.getindex(status::GitStatus, i::Csize_t)
-    if length(status) == 0
+    if i < 1 || i > length(status)
         throw(BoundsError())
     end
     entry_ptr = ccall((:git_status_byindex, :libgit2), Ptr{Void},
                        (Ptr{Void}, Csize_t), status.ptr, i-1)
-    entry_ptr == C_NULL && return nothing
```

Comment:
It only returns `C_NULL` if the index `i-1` is bad, which I already check for above.